shared/util/saga.tsx

```diff
@@ -120,19 +130,60 @@ function* chainAction2Impl<Actions extends {readonly type: string}>(
 
 export const chainAction2: ChainAction2 = (chainAction2Impl as unknown) as any
 
+function* chainActionImpl<Actions extends {readonly type: string}>(
+  pattern: Types.Pattern<any>,
+  f: (action: Actions, logger: SagaLogger) => ChainActionReturn
+) {
+  // @ts-ignore
+  return yield Effects.takeEvery<TypedActions>(pattern as Types.Pattern<any>, function* chainActionHelper(
+    action: TypedActions
+  ) {
+    const sl = new SagaLogger(action.type as ActionType, f.name ?? 'unknown')
+    try {
+      // @ts-ignore
+      const toPut = yield Effects.call(f, action, sl)
+      // release memory
+      // @ts-ignore
+      action = undefined
+      if (toPut) {
+        const outActions: Array<TypedActions> = isArray(toPut) ? toPut : [toPut]
+        for (var out of outActions) {
+          if (out) {
+            yield Effects.put(out)
+          }
+        }
+      }
+      if (sl.isTagged) {
```

Comment:
Is the tag the thing you pass to the logger when you create it? if so, seems like this should always be tagged.

Reply:
this is only true if you call tag on the logger yourself